pkg/sidecar/sidecarpb/slicegateway_handler.go

```diff
@@ -66,30 +60,30 @@ func sliceGwSetInterClusterDscpConfig(dscpClass string) error {
 		portFilter = "--source-port 11194"
 	}
 
-	if interClusterDscpClass == dscpClsStringToDscpCodeMap[dscpClass] {
-		log.Infof("No change in DSCP marking needed: %v", interClusterDscpClass)
+	if dscpClass == dscpClassInUse {
+		log.Infof("No change in DSCP marking needed: %v", dscpClassInUse)
 		return nil
 	}
 	// Delete existing DSCP config before adding a new one
-	if interClusterDscpClass != "Default" {
+	if dscpClassInUse != "Default" {
 		ipTablesCmd := fmt.Sprintf("iptables -t mangle -D POSTROUTING -p udp %s -j DSCP --set-dscp-class %s",
-			portFilter, interClusterDscpClass)
+			portFilter, dscpClassInUse)
 		_, err := runCommand(ipTablesCmd)
 		if err != nil {
-			log.Errorf("Could not remove existing DSCP config: %v. DSCP class in use: %v", err, interClusterDscpClass)
+			log.Errorf("Could not remove existing DSCP config: %v. DSCP class in use: %v", err, dscpClassInUse)
 			return err
 		}
 	}
```

Comment:
Please set dscpClassInUse to "Default" after the deletion is successful.

Reply:
I don't understand why do we need do this?

Reply:
Two main reasons:
dscpClassInUse must reflect the dscp class in use at any given point of time. 

1. Before any dscp marking is done, the class is "Default". Here after the deletion, we are back to square one with no marking, so the class in use must be set to "Default".
2. The delete succeeds but for whatever reason the add operation below fails. If the worker operator retries by sending the message again, you would never come to the add section. You will always error out trying to delete the dscp marking which is already deleted and does not exist. Below condition will always be true if dscpClassInUse holds a value that does not reflect the dscp marking in use, so if it is not set to "Default" you are stuck in a loop until the pod restarts.

if dscpClassInUse != "Default" {
		ipTablesCmd := fmt.Sprintf("iptables -t mangle -D POSTROUTING -p udp %s -j DSCP --set-dscp-class %s",
			portFilter, interClusterDscpClass)
			portFilter, dscpClassInUse)
		_, err := runCommand(ipTablesCmd)
		if err != nil {
			log.Errorf("Could not remove existing DSCP config: %v. DSCP class in use: %v", err, interClusterDscpClass)
			log.Errorf("Could not remove existing DSCP config: %v. DSCP class in use: %v", err, dscpClassInUse)
			return err
		}
	}